src/main/java/school/lemon/changerequest/java/io/hw1/IFileManagerImp.java

```diff
@@ -0,0 +1,343 @@
+                    break;
+                case BY_LEXICOGRAPHICAL_DESCENDING:
+                    Arrays.sort(paths, comparatorForLexDescending);
+                    for (File file : paths) {
+                        result.add(file);
+                    }
+                    break;
+                case BY_LAST_UPDATE_DESCENDING:
+                    Arrays.sort(paths, comparatorForLastUpdateDescending);
+                    for (File file : paths) {
+                        result.add(file);
+                    }
+                    break;
+                case BY_LAST_UPDATE_ASCENDING:
+                    Arrays.sort(paths, comparatorForLastUpdateAscending);
+                    for (File file : paths) {
+                        result.add(file);
+                    }
+                    break;
+            }
+        } else {
+            switch (order) {
+                case BY_LEXICOGRAPHICAL_ASCENDING:
+                    Arrays.sort(paths, comparatorForLexAscendingNoOrder);
+                    for (File file : paths) {
+                        result.add(file);
+                    }
+                    break;
+                case BY_LEXICOGRAPHICAL_DESCENDING:
+                    Arrays.sort(paths, comparatorForLexDescendingNoOrder);
+                    for (File file : paths) {
+                        result.add(file);
+                    }
+                    break;
+                case BY_LAST_UPDATE_DESCENDING:
+                    Arrays.sort(paths, comparatorForLastUpdateDescendingNoOrder);
+                    for (File file : paths) {
+                        result.add(file);
+                    }
+                    break;
+                case BY_LAST_UPDATE_ASCENDING:
+                    Arrays.sort(paths, comparatorForLastUpdateAscendingNoOrder);
+                    for (File file : paths) {
+                        result.add(file);
+                    }
+                    break;
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public String fileInfo(File file) {
+        if (!file.exists()) {
+            return null;
+        }
+        long size;
+        String fileName;
+        if (file.isDirectory()) {
+            size = 0;
+        } else {
+            size = file.length();
+        }
+        SimpleDateFormat sdf = new SimpleDateFormat("dd.MM.yyyy HH:mm:ss");
+        String lastModifiedDate = sdf.format(file.lastModified());
+        if (file.isDirectory()) fileName = file.getName() + File.separator;
+        else fileName = file.getName();
+        return String.format("%-20s|%15s|%s %n", lastModifiedDate, size, fileName);
+    }
+
+    @Override
+    public void createFile(String name) throws FileAlreadyExistException, FileManagerException {
+        File file = new File(name);
+        if (file.exists()) throw new FileAlreadyExistException(file);
+        try {
+            file.createNewFile();
+        } catch (IOException e) {
```

Comment:
What about IO exceptions? Why do you just ignore them?

You should wrap them into FileManagerException.